### DISCUSSION TITLE
Problem with hihglight code in dif comparison mode for python

### DISCUSSION THREAD

**Discussion by ManPython** [microsoft/pylance-release#2922] [Enhancement]
## Environment data
Pylance v2021.12.2
Intelisence v2021.12.1559732655
Python Coding Conventions v0.0.4
Python Path v0.0.11
Python Test Explorer for Visual Studio Code v0.7.0
PySide2-VSC v0.1.0

-   Language Server version: Python (right?)
-   OS and version: W10
-   Python version (& distribution if applicable, e.g. Anaconda): P3.8

## Expected behaviour
I copying this issue, due to I'm not sure it's pylance problem or other module from extensions for pyton
https://github.com/microsoft/vscode/issues/140067

## Actual behaviour
https://github.com/microsoft/vscode/issues/140067

## Logs

```
XXX
```

## Code Snippet / Additional information

```python
XXX
```

**Comment by ManPython** [comment]
There is related bug thats no changes in file expecting commit files to GIT in case some modify and back! (In fact no changes)
vsc 1.64.2

![bugs_git](https://user-images.githubusercontent.com/25403861/154821169-85c6fb26-a46c-4fea-a87f-3ab25faa19e7.png)

**Comment by judej** [comment]
Thanks for the report. Will investigate this.

**Comment by heejaechang** [comment]
we currently can't handle out of context single file for semantic info. it can only be done if we can clone cheaply with old files which we currently can't. for now, current behavior is by design.

**Comment by heejaechang** [comment]
one thing we could do as enhancement is creating open file workspace for the virtual document so we can provide semantic info at least for symbols defined in the file itself or symbols from library/stdlib packages

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
